core/src/main/scala/org/apache/spark/scheduler/CookSchedulerContext.scala

```diff
@@ -109,6 +117,20 @@ case class CookSchedulerContext(
         tgz.split(",").map(_.trim).toList
       }
 
+  /**
+    * Parse the Cook constraints from Spark configurations.
+    *
+    * The Cook constraint(s) could be provided as follows.
+    *  - spark.cook.job.constraint=[foo,EQUALS,bar]
+    *  - spark.cook.job.constraint.1=[foo1,EQUALS,bar1]
+    *  - spark.cook.job.constraint.2=[foo2,EQUALS,bar2]
+    */
+  val cookJobConstraints: Seq[Constraint] = conf.getAll.filter { case (key, _) =>
+    key.matches("^spark\\.cook\\.job\\.constraint(\\.\\d+$|$)")
+  }.map { case (_, constraint) =>
+    Constraints.parseFrom(new JSONArray(constraint))
```

Comment:
Is `JSONArray` going to parse `[foo1,EQUALS,bar1]` or do those need to be quoted like this `["foo1","EQUALS","bar1"]`?

Reply:
You don't need to quote.  See https://github.com/twosigma/Cook/blob/6c9c0451244fffeabfbe9eca2f5e34ef847621e3/jobclient/src/test/java/com/twosigma/ConstraintTest.java#L37